spec/game_setting_setter_spec.rb

```diff
@@ -0,0 +1,47 @@
+require_relative '../lib/game_setting/game_setting'
+require_relative '../lib/game_setting/opponent_type'
+require_relative '../lib/game_setting/board_size'
+require_relative '../lib/player/player_types'
+require_relative '../lib/game_setting_setter'
+require_relative '../lib/game_presenter/test_io'
+
+describe 'Game Setting Setter' do
+    include PlayerType
+    
+    attr_accessor :game_setting_setter
+
+    setting_types = [
+        GameSetting.new(setting: BoardSize.new()),
+        GameSetting.new(setting: OpponentType.new()),
+    ]
+
+    before(:all) do
```

Comment:
I would try to use `before(:each)` as much as possible instead of `before(:all)`. It helps keep your tests isolated